### PR TITLE
🌱Remove unused fields from ExternalNetwork

### PR DESCRIPTION
**Pull request by mdbooth**
OpenStackClusterStatus.ExternalNetwork currently shares a type with OpenStackClusterStatus.Network, but the uses are not similar. It should have its own type.

The only field which is currently used is ID. However, Name and Tags have been present for a long time and could reasonably be informational.

Fixes: #1552 

/hold

### CODE REVIEW

File: api/v1alpha7/types.go [hunk 1]
```diff
@@ -197,6 +197,15 @@ type Network struct {
 	APIServerLoadBalancer *LoadBalancer `json:"apiServerLoadBalancer,omitempty"`
 }
 
+// NetworkStatus contains basic information about an existing neutron network.
+type NetworkStatus struct {
```
Comment (dulek) [inline api/v1alpha7/types.go:201]: Is it really a `Status`? Feels more like a `Spec` to me.

Comment (mdbooth) [reply api/v1alpha7/types.go:201]: It's definitely a Status: it has an ID apart from anything else. In the Spec we use NetworkFilter.

**Review by seanschneeweiss — COMMENTED** [review]
/lgtm

File: api/v1alpha7/openstackcluster_types.go [hunk 2]
```diff
@@ -159,8 +159,8 @@ type OpenStackClusterStatus struct {
 	// It includes Subnets and Router.
 	Network *Network `json:"network,omitempty"`
 
-	// External Network contains information about the created OpenStack external network.
-	ExternalNetwork *Network `json:"externalNetwork,omitempty"`
+	// externalNetwork contains information about the external network used for default ingress and egress traffic.
```
Comment (jichenjc) [inline api/v1alpha7/openstackcluster_types.go:162]: s/externalNetwork /ExternalNetwork/

**Pull request merged**